### ISSUE TITLE
Older direct links stop working

### ISSUE THREAD

**Issue by xaur** [t3chguy/matrix-static#65]
I opened this message direct link from Jul 5

https://view.matrix.org/room/!wSdymYrEpBhsWlDJuk:decred.org/?anchor=$153090455433548QNlFj:decred.org

and it showed "Some error has occurred". Then I removed the anchor part from the URL to open latest messages in the room and pressed "Load older messages" until I reached Jul 5 and it showed the message I was after. Then I visited initial direct link again and it opened.

Example of direct link that doesn't work (until someone visits it via "Load older messages"): https://view.matrix.org/room/!OfChXgczrIlpEZSFAv:decred.org/?anchor=$153125687135698lmQRO:decred.org

**Comment by t3chguy** [comment]
Yes because of the limitations of the peeking APIs it cannot access old events without having to paginate to them which would be very expensive

**Comment by xaur** [comment]
Does it mean a won't fix?

**Comment by t3chguy** [comment]
It'd mean having to persist things locally (duplicating data) but it's not completely out of the question

**Comment by xaur** [comment]
Thanks. I guess I shall be using matrix.to "dispatch" links then -- tried opening some "old" links in Riot and they open pretty fast.

**Comment by t3chguy** [comment]
Riot uses the context api which is relatively expensive and would have to be done for each event, matrix-static pulls pages of events in order and already runs into issues with hitting the server too much

**Comment by bkil** [comment]
Related: #115
